idx/cassandra/cassandra.go

```diff
@@ -383,6 +390,22 @@ NAMES:
 	return defs
 }
 
+func (c *CasIdx) ArchiveDefs(defs []schema.MetricDefinition) error {
+	for _, def := range defs {
```

Comment:
processing each def sequentially will be really slow.  Lets use a pool of goroutines to speed this up.
```
func (c *CasIdx) ArchiveDefs(defs []schema.MetricDefinition) error {
	defChan := make(chan *schema.MetricDefinition, c.cfg.numConns)
	g, ctx := errgroup.WithContext(context.Background())
	for i := 0; i < c.cfg.numConns; i++ {
		g.Go(func() error {
			for {
				select {
				case def, ok := <-defChan:
					if !ok {
						return nil
					}
					err := c.addDefToArchive(*def)
					if err != nil {
						return err
					}

					err = c.deleteDef(def.Id, def.Partition)
					if err != nil {
						return err
					}
				case <-ctx.Done():
					return ctx.Err()
				}
			}
			return nil
		})
	}
	for i := range defs {
		defChan <- &defs[i]
	}
	close(defChan)
	if err := g.Wait(); err != nil {
		return err
	}

	return nil
}
```

Reply:
i was actually considering that at the time when i wrote it, then i figured there is usually not a lot of time pressure when executing this so there's no need to put more load on cassandra and potentially slow down operations that actually affect users